classes/fetch_services.py

```diff
@@ -80,6 +81,9 @@ def parse(self, response) -> dict:
             "duration": convert_iso8601_duration_to_seconds(iso8601_duration),
         }
 
+def prompt_for_missing_durs(prompt_if_missing):
+    global prompt_dur
```

Comment:
I would discourage the use of `global` as it makes code more difficult to debug (since it's difficult to trace where and when a global variable gets set).

An alternative way to pass the `prompt_dur` information to the class is as an extra argument to the `__init__` function of `YtDlpFetchService`, like this:

    def __init__(self, accepted_domains: list[str], prompt_if_dur_missing: bool=false):
        self.accepted_domains = accepted_domains
        self.prompt_if_dur_missing = prompt_if_dur_missing

This avoids the use of global variables and makes it easier to track the flow of information through the application.